src/com/kodilla/controls/BoardPresenter.java

```diff
@@ -1,24 +1,49 @@
 package com.kodilla.controls;
 
+import com.kodilla.controls.winConditions.ColumnWinChecker;
+import com.kodilla.controls.winConditions.DiagonalWinChecker;
+import com.kodilla.controls.winConditions.RowWinChecker;
 import com.kodilla.ui.Board;
+import javafx.scene.control.Alert;
+import javafx.scene.control.ButtonType;
+import javafx.stage.Stage;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
-public class BoardPresenter implements Connector{
+public class BoardPresenter implements Connector.Presenter {
 
-    private final List<FieldState> gameBoard;
     private final BoardSettings settings;
+    private boolean ifWin;
+    private List<FieldState> gameBoard;
+    private Board view;
```

Comment:
Simple `Board` is not enough, Board sounds like a model object

Reply:
done!